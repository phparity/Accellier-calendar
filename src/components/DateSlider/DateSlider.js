import React, { useState, useEffect, Fragment } from "react";
import { useSwipeable } from "react-swipeable";
import DateCard from "../DateCard/DateCard";
import LeftArrowIcon from "../../assets/images/arrow-left.svg";
import RightArrowIcon from "../../assets/images/arrow-right.svg";
import "./DateSlider.scss";
import PostCard from "../PostCard/PostCard";
import PostCardList from "../PostCard/PostCardList";

/**
 * Generates a date slider component with the ability to navigate through dates and display events for each date.
 *
 * @param {object} selectedDate - The selected date for the date slider.
 * @param {array} eventListData - The list of events to display on the date slider.
 * @return {JSX.Element} The date slider component.
 */
const DateSlider = ({ selectedDate, eventListData }) => {
  // Initialize currentDate state with selectedDate if it exists, otherwise use current date
  const [currentDate, setCurrentDate] = useState(
    selectedDate ? new Date(selectedDate) : new Date()
  );
  // This hook initializes the state variable daysPerPage with a default value of 7
  const [daysPerPage, setDaysPerPage] = useState(7);

  // useEffect hook to update current date when selectedDate changes
  useEffect(() => {
    if (selectedDate) setCurrentDate(selectedDate);
  }, [selectedDate]);

  /**
   * Returns the day of the week based on the day index.
   * @param {number} dayIndex - The index of the day in the week.
   * @returns {string} - The name of the day.
   */
  const getDayOfWeek = (dayIndex) => {
    // Array of days in a week
    const daysOfWeek = [
      "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"
    ];
    // Return the day based on the index
    return daysOfWeek[dayIndex % daysOfWeek.length];
  };

  /**
   * Returns the name of the month based on the given month index.
   * @param {number} monthIndex - The index of the month (0-11).
   * @returns {string} - The name of the month.
   */
  const getMonthName = (monthIndex) => {
    // Array of month names
    const months = [
      "January", "February", "March", "April",
      "May", "June", "July", "August",
      "September", "October", "November", "December"
    ];
    // Return the month name based on the index
    return months[monthIndex % months.length];
  };
  /**
   * Renders the days and corresponding post cards for the current date and surrounding dates.
   * @returns {Object} An object containing arrays of days and post cards.
   */
  const renderDays = () => {
    // Initialize arrays to store days and post cards
    const days = [];
    const postCards = [];
    // Calculate the offset based on the number of days per page
    const offset = Math.floor(daysPerPage / 2);
    // Iterate through the surrounding dates
    for (let i = -offset; i <= offset; i++) {
      // Create a new date object for the current date and offset
      const date = new Date(currentDate);
      date.setDate(currentDate.getDate() + i);
      // Check if the current date is active
      const isActive =
        date.getDate() === new Date().getDate() &&
        date.getMonth() === new Date().getMonth() &&
        date.getFullYear() === new Date().getFullYear();
      // Create a DateCard component and add it to the days array
      days.push(
        <DateCard
          key={i + offset}
          DateText={`${getMonthName(
            date.getMonth()
          )} ${date.getDate()}, ${date.getFullYear()}`}
          DayText={getDayOfWeek(date.getDay())}
          className={isActive ? "current_date" : ""}
        />
      );
      // Filter the event list data for the current date
      const postCardData = eventListData.filter(
        (post) =>
          post.date ===
          `${date.getMonth() + 1}/${date.getDate()}/${date.getFullYear()}`
      );
      // Create post card components and add them to the post cards array
      if (postCardData.length > 0) {
        postCards[i + offset] = postCardData.map((postCard, index) => (
          <Fragment key={`${postCard.id}_${index}`}>
            <PostCard PostListData={postCard} />
          </Fragment>
        ));
      } else {
        postCards[i + offset] = (
          <Fragment key={`empty_${i + offset}`}>
            <PostCard PostListData={""} />
          </Fragment>
        );
      }
    }
    // Return the arrays of days and post cards
    return { days, postCards };
  };

  /**
   * Update the current date by adding a specified number of days.
   * @param {number} days - The number of days to add to the current date.
   */
  const handleSwipe = (days) => {
    // Calculate the new date by adding the specified number of days to the current date
    setCurrentDate(new Date(currentDate.getTime() + days * 24 * 60 * 60 * 1000));
  };
  // Set up swipe handlers
  // const handlers = useSwipeable({
  //   onSwipedLeft: () => handleSwipe(1), // Call handleSwipe with 1 when swiped left
  //   onSwipedRight: () => handleSwipe(-1), // Call handleSwipe with -1 when swiped right
  //   preventDefaultTouchmoveEvent: true, // Prevent default touchmove event
  //   trackMouse: true, // Enable tracking of mouse events
  //   delta: 30,                             // min distance(px) before a swipe starts. *See Notes*
  //   trackTouch: true,                      // track touch input
  //   swipeDuration: Infinity,               // allowable duration of a swipe (ms). *See Notes*
  //   touchEventOptions: { passive: true },
  // });
  const handlers = useSwipeable({
    onSwipedLeft: () => handleSwipe(1), // Call handleSwipe with 1 when swiped left
    onSwipedRight: () => handleSwipe(-1), // Call handleSwipe with -1 when swiped right
    preventDefaultTouchmoveEvent: true, // Prevent default touchmove event
    trackMouse: true, // Enable tracking of mouse events
    delta: 10,                             // min distance(px) before a swipe starts. *See Notes*
    swipeDuration: Infinity,               // allowable duration of a swipe (ms). *See Notes*
    touchEventOptions: { passive: true },
    swipeDuration: 500,
    preventScrollOnSwipe: true,
    trackMouse: true
  });

  useEffect(() => {
    /*
    * updateColumns - updates the number of days per page based on the screen width
    */
    const updateColumns = () => {
      const screenWidth = window.innerWidth;
      // if (screenWidth >= 1800) {     
      //   setDaysPerPage(7); // set 7 days per page for large screens
      // } else if (screenWidth >= 1700) { 
      //   setDaysPerPage(5); // set 5 days per page for medium screens
      // } else if (screenWidth >= 1340) {
      //   setDaysPerPage(4); // set 4 days per page for medium screens 
      // }else if (screenWidth >= 992) {
      //   setDaysPerPage(3); // set 3 days per page for small screens
      // } else if (screenWidth >= 767) {
      //   setDaysPerPage(2); // set 2 days per page for small screens
      // } else {
      //   setDaysPerPage(1); // set 1 day per page for extra small screens
      // } 
      if (screenWidth >= 1250) {
        setDaysPerPage(7); // set 7 days per page for large screens
      } else if (screenWidth >= 1024) {
        setDaysPerPage(5); // set 5 days per page for medium screens
      } else if (screenWidth >= 667) {
        setDaysPerPage(3); // set 3 days per page for small screens
      } else {
        setDaysPerPage(1); // set 1 day per page for extra small screens
      }
    };
    // Call updateColumns function
    updateColumns();
    // Add event listener to update columns on window resize
    window.addEventListener("resize", updateColumns);
    // Remove event listener when component is unmounted
    return () => {
      window.removeEventListener("resize", updateColumns);
    };
  }, []);

  return (
    <div className="management_dates" {...handlers}>
      <div className="date_cards_otr">{renderDays().days}</div>
      <div className="arrows_otr">
        <div
          className="arrow_otr left_arrow"
          onClick={() => handleSwipe(-1)}
        >
          <img className="arrow_icon" src={LeftArrowIcon} alt="icon" />
        </div>
        <div
          className="arrow_otr right_arrow"
          onClick={() => handleSwipe(1)}
        >
          <img className="arrow_icon" src={RightArrowIcon} alt="icon" />
        </div>
      </div>

      <div className="post_row_custom">
        {Object.keys(renderDays().postCards).map((dayIndex) => (
          <div key={dayIndex} className="post_cards_otr">
            {renderDays().postCards[dayIndex]}
          </div>
        ))}
      </div>
    </div>
  );
};

export default DateSlider;
