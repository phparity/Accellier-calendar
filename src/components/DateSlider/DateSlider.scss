@import '../../assets/common-style/variable.scss';



.management_dates {
    position: relative;
    // background-color: $Light-Bg2;
    background-color: #F3F4F9;
    @media (max-width:770px){
        border-top: 1px solid #C2C2C2;
        border-bottom: 1px solid #C2C2C2;
        .arrows_otr{
            display: flex;
            align-items: center;
            justify-content: space-between;
            position: absolute;
            left: 50%;
            top: 14px;
            transform: translateX(-50%);
            -webkit-transform: translateX(-50%);
            -moz-transform: translateX(-50%);
            -ms-transform: translateX(-50%);
            -o-transform: translateX(-50%);
            width: 220px;
            .arrow_otr{
                position: initial;
                top: auto !important;
                left: auto !important;
                right: auto !important;
                transform: translateY(0);
                -webkit-transform: translateY(0);
                -moz-transform: translateY(0);
                -ms-transform: translateY(0);
                -o-transform: translateY(0);
            }
        }
    }
    .arrow_otr {
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        top: 2.4em;
        transform: translateY(-50%);
        -webkit-transform: translateY(-50%);
        -moz-transform: translateY(-50%);
        -ms-transform: translateY(-50%);
        -o-transform: translateY(-50%);
        width: 24px;
        height: 24px;
        background-color: $Primary;
        border-radius: 100%;
        -webkit-border-radius: 100%;
        -moz-border-radius: 100%;
        -ms-border-radius: 100%;
        -o-border-radius: 100%;
        transition: .3s;
        -webkit-transition: .3s;
        -moz-transition: .3s;
        -ms-transition: .3s;
        -o-transition: .3s;
        cursor: pointer;
        @media (max-width:1700px){
            width: 18px;
            height: 18px;
        }
        @media (max-width:770px){
            width: 14.26px;
            height: 14.26px;
        }
        &:hover {
            background-color: $Black;
        }
        &.left_arrow {
            left: 24px;
            @media (max-width:1700px){
                left: 18px;
            }
        }
        &.right_arrow {
            right: 19px;
            @media (max-width:1700px){
                right: 14.4px;
            }
        }
        &[disabled] {
            opacity: 25%;
            pointer-events: none;
        }
        .arrow_icon {
            width: 18.58px;
            height: 18.58px;
            object-fit: contain;
            object-position: center;
            @media (max-width:1700px){
                width: 14px;
                height: 14px;
            }
            @media (max-width:770px){
                width: 11.04px;
                height: 11.04px;
            }
        }
    }
}