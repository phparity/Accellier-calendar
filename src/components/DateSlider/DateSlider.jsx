/* eslint-disable react/prop-types */


import LeftArrowIcon from "../../../public/images/arrow-left.svg";
import RightArrowIcon from "../../../public/images/arrow-right.svg";

import "./DateSlider.scss";

const DateSlider = ({handlePrev,handleNext,currentPage,daysPerPage}) => {

    const currentDate = new Date();
    const currentMonth = currentDate.getMonth();
    const currentYear = currentDate.getFullYear();
    const totalDaysInMonth = new Date(currentYear, currentMonth + 1, 0).getDate();

  return (
    <div style={{  zIndex: 2 }} className="management_dates ">
      <div className="arrows_otr">
        <div
          className="arrow_otr left_arrow"
            onClick={handlePrev}
            disabled={currentPage === 0}
        >
          <img className="arrow_icon" src={LeftArrowIcon} alt="icon" />
        </div>
        <div
          className="arrow_otr right_arrow"
            onClick={handleNext}
            disabled={
              currentPage === Math.ceil(totalDaysInMonth / daysPerPage) - 1
            }
        >
          <img className="arrow_icon" src={RightArrowIcon} alt="icon" />
        </div>
      </div>
    </div>
  );
};

export default DateSlider;
