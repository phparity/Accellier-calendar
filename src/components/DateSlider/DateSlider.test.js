import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import DateSlider from "./DateSlider";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

// Mock eventListData
const mockEventListData = [
  { id: 1, date: "2/8/2024" },
  { id: 2, date: "2/9/2024" },
  { id: 3, date: "2/10/2024" },
];

describe("DateSlider Component", () => {
  test("DateSlider renders properly and navigates correctly", () => {
    // Render the DateSlider component
    render(
      <Router>
        <DateSlider
          selectedDate={new Date("2024-02-08")}
          eventListData={mockEventListData}
        />
      </Router>,
    );

    // Check if date cards are rendered
    const dateCard = screen.getByTestId("date-card");
    expect(dateCard).toBeInTheDocument();

    // Check if post cards are rendered
    const postCard = screen.getByTestId("post-card");
    expect(postCard).toBeInTheDocument();

    // Check if left arrow is rendered
    const leftArrow = screen.getByTestId("left-arrow");
    expect(leftArrow).toBeInTheDocument();

    // Check if right arrow is rendered
    const rightArrow = screen.getByTestId("right-arrow");
    expect(rightArrow).toBeInTheDocument();

    // Simulate clicking on the left arrow
    fireEvent.click(leftArrow);

    // Simulate clicking on the right arrow
    fireEvent.click(rightArrow);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <DateSlider
            selectedDate={new Date("2024-02-08")}
            eventListData={mockEventListData}
          />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
