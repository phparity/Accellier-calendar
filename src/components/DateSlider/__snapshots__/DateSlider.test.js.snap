// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSlider Component renders correctly 1`] = `
<div
  className="management_dates"
  onMouseDown={[Function]}
>
  <div
    className="date_cards_otr"
    data-testid="date-card"
  >
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 5, 2024
        </h3>
        <p
          className="day_text"
        >
          Monday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 6, 2024
        </h3>
        <p
          className="day_text"
        >
          Tuesday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 7, 2024
        </h3>
        <p
          className="day_text"
        >
          Wednesday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 8, 2024
        </h3>
        <p
          className="day_text"
        >
          Thursday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 9, 2024
        </h3>
        <p
          className="day_text"
        >
          Friday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 10, 2024
        </h3>
        <p
          className="day_text"
        >
          Saturday
        </p>
      </div>
    </div>
    <div
      className="date_card_otr "
      data-testid="date-card-outer"
    >
      <div
        className="date_card_inr"
      >
        <h3
          className="date_text"
        >
          February 11, 2024
        </h3>
        <p
          className="day_text"
        >
          Sunday
        </p>
      </div>
    </div>
  </div>
  <div
    className="arrows_otr"
  >
    <div
      className="arrow_otr left_arrow"
      data-testid="left-arrow"
      onClick={[Function]}
    >
      <img
        alt="left-icon"
        className="arrow_icon"
        src="test-file-stub"
      />
    </div>
    <div
      className="arrow_otr right_arrow"
      data-testid="right-arrow"
      onClick={[Function]}
    >
      <img
        alt="right-icon"
        className="arrow_icon"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="post_row_custom"
    data-testid="post-card"
  >
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr empty-post"
      />
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr empty-post"
      />
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr empty-post"
      />
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr undefined"
      >
        <div
          className="card_header"
          data-testid="card-header"
          onClick={[Function]}
        >
          <div
            className="heading_tag"
          >
            <h3
              className="card_heading"
            >
              <a
                href="/eventmanagement"
                onClick={[Function]}
              >
                undefined2/8/2024
              </a>
            </h3>
          </div>
          <div
            className="location_time"
          >
            <div
              className="icon_text_otr"
            >
              <img
                alt="time-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
            <div
              className="icon_text_otr location_text_otr"
            >
              <img
                alt="location-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
          </div>
        </div>
        <div
          className="card_body "
          data-testid="card-body"
        >
          <div
            className="profile_otr"
          >
            <img
              alt="img"
              className="profile_img"
            />
            <div
              className="profile_text"
            >
              <p
                className="user_name"
              />
              <p
                className="user_post"
              />
            </div>
          </div>
          <div
            className="info_icons_otr"
          >
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip1"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip2"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip3"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
          </div>
          <div
            className="task_equipments_otr"
          >
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/eventmanagement"
                  onClick={[Function]}
                >
                  Tasks
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Equipment
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Inventory
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Staffing
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="sales_profit"
          >
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
            <span
              className="line"
            />
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr undefined"
      >
        <div
          className="card_header"
          data-testid="card-header"
          onClick={[Function]}
        >
          <div
            className="heading_tag"
          >
            <h3
              className="card_heading"
            >
              <a
                href="/eventmanagement"
                onClick={[Function]}
              >
                undefined2/9/2024
              </a>
            </h3>
          </div>
          <div
            className="location_time"
          >
            <div
              className="icon_text_otr"
            >
              <img
                alt="time-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
            <div
              className="icon_text_otr location_text_otr"
            >
              <img
                alt="location-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
          </div>
        </div>
        <div
          className="card_body "
          data-testid="card-body"
        >
          <div
            className="profile_otr"
          >
            <img
              alt="img"
              className="profile_img"
            />
            <div
              className="profile_text"
            >
              <p
                className="user_name"
              />
              <p
                className="user_post"
              />
            </div>
          </div>
          <div
            className="info_icons_otr"
          >
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip1"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip2"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip3"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
          </div>
          <div
            className="task_equipments_otr"
          >
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/eventmanagement"
                  onClick={[Function]}
                >
                  Tasks
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Equipment
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Inventory
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Staffing
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="sales_profit"
          >
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
            <span
              className="line"
            />
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr undefined"
      >
        <div
          className="card_header"
          data-testid="card-header"
          onClick={[Function]}
        >
          <div
            className="heading_tag"
          >
            <h3
              className="card_heading"
            >
              <a
                href="/eventmanagement"
                onClick={[Function]}
              >
                undefined2/10/2024
              </a>
            </h3>
          </div>
          <div
            className="location_time"
          >
            <div
              className="icon_text_otr"
            >
              <img
                alt="time-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
            <div
              className="icon_text_otr location_text_otr"
            >
              <img
                alt="location-icon"
                className="icon"
                src="test-file-stub"
              />
              <p
                className="icon_text"
              />
            </div>
          </div>
        </div>
        <div
          className="card_body "
          data-testid="card-body"
        >
          <div
            className="profile_otr"
          >
            <img
              alt="img"
              className="profile_img"
            />
            <div
              className="profile_text"
            >
              <p
                className="user_name"
              />
              <p
                className="user_post"
              />
            </div>
          </div>
          <div
            className="info_icons_otr"
          >
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip1"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip2"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
            <div
              className="info_icon_inr"
            >
              <div
                className="icon_tooltip icon_tooltip3"
              >
                <p
                  className="tooltip_text"
                />
              </div>
              <img
                alt="icon"
                className="icon_img"
              />
            </div>
          </div>
          <div
            className="task_equipments_otr"
          >
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/eventmanagement"
                  onClick={[Function]}
                >
                  Tasks
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Equipment
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Inventory
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
            <div
              className="task_otr content_otr"
            >
              <p
                className="task_text"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Staffing
                </a>
              </p>
              <div
                className="task_row"
              >
                <div
                  className="task_num_inr"
                >
                  <p
                    className="red_arrow_num arrow_num"
                  />
                  <p
                    className="yellow_arrow_num arrow_num"
                  />
                  <p
                    className="green_arrow_num arrow_num"
                  />
                </div>
                <div
                  className="task_arrow_inr"
                >
                  <ul
                    className="red_arrow"
                  />
                  <ul
                    className="yellow_arrow"
                  />
                  <ul
                    className="green_arrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="sales_profit"
          >
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
            <span
              className="line"
            />
            <div
              className="forcast_otr"
            >
              <p
                className="forcast_text"
              >
                Forecast Sales
              </p>
              <p
                className="sales_money"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="post_cards_otr"
    >
      <div
        className="post_card_inr empty-post"
      />
    </div>
  </div>
</div>
`;
